hack/clean/clean.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"flag"
+	"os"
+	"strings"
+	"time"
+
+	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
+	"github.com/sirupsen/logrus"
+
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+	"github.com/Azure/ARO-RP/pkg/util/purge"
+)
+
+const (
+	defaultTTL     = 48
+	defaultKeepTag = "keepit"
+)
+
+var dryRun = flag.Bool("dryRun", false, "should the purge perform dry for test")
+
+func main() {
+	ctx := context.Background()
+	log := utillog.GetLogger()
+
+	flag.Parse()
+
+	if err := run(ctx, log); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(ctx context.Context, log *logrus.Entry) error {
+	var ttl time.Duration
+	if os.Getenv("AZURE_PURGE_TTL") != "" {
+		var err error
+		ttl, err = time.ParseDuration(os.Getenv("AZURE_PURGE_TTL"))
+		if err != nil {
+			return err
+		}
+	} else {
+		ttl = defaultTTL
+	}
+
+	deleteCheck := func(resourceGroup mgmtfeatures.ResourceGroup, log *logrus.Entry) bool {
```

Comment:
style - why is this a closure?

Reply:
could you call the function "shouldDelete" or something like that to make its sense clearer?

Reply:
My thinking for the closure is the exposure.

I wanted to expose the check directly to the main code. Then everyone who edits the code directly sees how the check is performed. No need to dig into the dependencies.

Moreover, the check can be more complicated in the next versions. Meaning, closure is a better choice than multiple conditions controllers by parameters.

Reply:
I think either way people will need to dig into code. Leta not try to remove the need to think from people writing code :)